app/services/taxonomy/find_childest.rb

```diff
@@ -0,0 +1,25 @@
+module Taxonomy
+  class FindChildest
```

Comment:
There is also a module `Taxonomy` defined in `lib/taxonomy.rb` so maybe this could live there?